tools/src/main/scala/scala/scalanative/build/Discover.scala

```diff
@@ -90,6 +90,25 @@ object Discover {
     }
   }
 
+  /** Detect the target architecture as a Scala Native build structure
+   * @param triple the LLVM target triple, see Discover.targetTriple
+   * @return the detected target architecture
+   */
+  def targetArchitecture(triple: String): TargetArchitecture = {
+    import TargetArchitecture._
+    triple.split('-').head match {
+      case "x86_64"  => x86_64
+      case "i386"    => i386
+      case "i686"    => i686
+      case "armv7l"  => arm
```

Comment:
If we're using LLVM target architecture names, it's probably best to keep them as-is (i.e. `armv7l` on the right here). Otherwise there is a chance of confusing name clashes.